powermax/provider/snapshotpolicy_resource.go

```diff
@@ -0,0 +1,455 @@
+/*
+Copyright (c) 2022-2023 Dell Inc., or its subsidiaries. All Rights Reserved.
+
+Licensed under the Mozilla Public License Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://mozilla.org/MPL/2.0/
+
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	pmax "dell/powermax-go-client"
+	"fmt"
+	"regexp"
+	"terraform-provider-powermax/client"
+	"terraform-provider-powermax/powermax/constants"
+	"terraform-provider-powermax/powermax/helper"
+	"terraform-provider-powermax/powermax/models"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64default"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &SnapshotPolicy{}
+var _ resource.ResourceWithImportState = &SnapshotPolicy{}
+var _ resource.ResourceWithConfigure = &SnapshotPolicy{}
+
+// NewSnapshotPolicy creates a new Snapshot Policy resource.
+func NewSnapshotPolicy() resource.Resource {
+	return &SnapshotPolicy{}
+}
+
+// SnapshotPolicy defines the resource implementation.
+type SnapshotPolicy struct {
+	client *client.Client
+}
+
+// Metadata returns the metadata for the resource.
+func (r *SnapshotPolicy) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_snapshotpolicy"
+}
+
+// Schema returns the schema for the resource.
+func (r *SnapshotPolicy) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Resource for a specific Snapshot Policy in PowerMax array.",
+		Description:         "Resource for a specific Snapshot Policy in PowerMax array.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Identifier",
+				Computed:    true,
+			},
+			"snapshot_policy_name": schema.StringAttribute{
+				Description:         "Name of the snapshot policy",
```

Comment:
@nikitajoshi1 provide the validation rule in description like name should be maximum 32 characters long etc.

Reply:
Updated Description.